### PR TITLE
pod.node: Raise if pod is not running

### PR DESCRIPTION
**Pull request by myakove**

### CODE REVIEW

**Review by rnetser — APPROVED** [review]

**Comment by myakove** [comment]
!verified

File: ocp_resources/pod.py [hunk 1]
```diff
@@ -161,6 +161,7 @@ def node(self):
         Returns:
             Node: Node
         """
+        assert self.instance.spec.nodeName, f"Node not found for pod {self.name}"
```
Comment (myakove) [inline ocp_resources/pod.py:164]: `self.instance.spec.nodeName` need to save and reuse.

**Comment by myakove** [comment]
!verified

**Review by rnetser — APPROVED** [review]

**Comment by myakove** [comment]
!cherry-pick v4.11

**Pull request merged**